### PR TITLE
Ft product review

### PR DESCRIPTION
**Pull request by louisdante9**
What does this PR do?

Add a product review and shows previous reviews if any

Description of Task to be completed?

A user should be able to post a product review and also see previous reviews on the current product being viewed 

How should this be manually tested?

Login, click on a "Product" from the list of product on the shop page, this redirects the user to a single product that shows previous ratings and reviews. rate the product under the submit a review section and post a review on the product 

What are the relevant pivotal tracker stories?

#151475461

Screenshots (if appropriate)
screen shot 2017-10-09 at 10 31 15 am

### CODE REVIEW

File: tests/acceptance-tests/test/specs/smoke-tests/review.app-test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const fs   = require("fs");
+const yaml = require("js-yaml");
+const expect = require("chai").expect;
+const getId = require("../../../lib/get-elements.js");
+
+beforeEach(function () {
+  const browserConfig = yaml.safeLoad(fs.readFileSync("./tests/acceptance-tests/config/settings.yml", "utf8"));
+  const baseUrl = browserConfig.base_url.toString();
+  browser.url(baseUrl);
+  // browser.getSession().then(function (sessionid) {
```
Comment (mradeybee) [inline tests/acceptance-tests/test/specs/smoke-tests/review.app-test.js]: Remove Dead Code

File: imports/plugins/included/product-detail-simple/client/templates/review.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    review.productId = Products.findOne(productId())._id;
+    try {
+      review.username = Meteor.user().username || Meteor.user().emails[0].address;
+      review.dateCreated = new Date;
+      Meteor.call("insert/review", review, function (error, result) {
+        console.log(error, result);
+        if (error) {
+          return error;
+        }
+        console.log(result);
```
Comment (mradeybee) [inline imports/plugins/included/product-detail-simple/client/templates/review.js]: Why are you logging result here?

File: imports/plugins/included/product-detail-simple/client/templates/review.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      Alerts.toast("You need to sign in to post a review", "error");
+    }
+  }
+});
+
+
+Template.showReviews.helpers({
+  reviews: () => {
+    const productId = () => Reaction.Router.getParam("handle");
+    console.log(productId());
```
Comment (mradeybee) [inline imports/plugins/included/product-detail-simple/client/templates/review.js]: why console.log?

File: server/publications/collections/reviews.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { Reviews } from "/lib/collections";
+import { Meteor } from "meteor/meteor";
+
+Meteor.publish("Reviews", function () {
+  return Reviews.find({});
+});
```
Comment (mradeybee) [inline server/publications/collections/reviews.js]: Add an empty line at the of the file

File: tests/acceptance-tests/elements/element-ids.yml [hunk 5]
```diff
@@ -5,4 +5,5 @@ search_input_id: search-input
 cardnumber_id: cardnumber
 expiry_id: expiry
 cvv_id: cvv
-pay_btn: pay-btn
\ No newline at end of file
+pay_btn: pay-btn
+comment: comment
```
Comment (mradeybee) [inline tests/acceptance-tests/elements/element-ids.yml]: Add an empty line at the end of the file.

File: tests/acceptance-tests/test/specs/smoke-tests/review.app-test.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    browser.click(eleMap.star_click);
+    browser.pause("1000");
+    browser.setValue("#comment", "This is a review");
+    browser.pause("2000");
+    browser.click(eleMap.send_review);
+    browser.pause("1000");
+  
+    expect(browser.getAttribute("div", "show-review")).to.exist;
+  });
+});
```
Comment (mradeybee) [inline tests/acceptance-tests/test/specs/smoke-tests/review.app-test.js]: Add an empty line at the end of the file.

File: imports/plugins/included/product-detail-simple/client/templates/review.html [hunk 7]
```diff
@@ -12,10 +12,39 @@
...
+            {{> starsRating rating=review.rating size=20}}
+          </div>
+          <div>
+            {{review.comment}}
+          </div>
+        </div>
+        <br> {{/each}} {{/if}}
+      </div>
+    </div>
+  </div>
```
Comment (mradeybee) [inline imports/plugins/included/product-detail-simple/client/templates/review.html:49]: Add an empty line at the end of the file.

File: imports/plugins/included/product-detail-simple/client/templates/productDetailSimple.html [hunk 8]
```diff
@@ -1,11 +1,10 @@
...
-</template>
+  {{#if isEnabled}} {{!-- Use new product detail page --}}
+  <div>
+    {{> React component=PDC}}
+  </div>
+  {{else}} {{!-- Use old product detail page --}} {{> productDetail }} {{/if}}
+  <div>
+    {{>showReviews}} {{>productReviews}}
+  </div>
+</template>
```
Comment (mradeybee) [inline imports/plugins/included/product-detail-simple/client/templates/productDetailSimple.html]: Add an empty line at the end of the file.

**Review by mradeybee — CHANGES_REQUESTED** [review]
I have few comments, nothing major.

File: imports/plugins/included/product-detail-simple/client/templates/review.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      return false;
+    }
+    const productId = () => Reaction.Router.getParam("handle");
+      document.getElementById("comment").value = "";
+    review.productId = Products.findOne(productId())._id;
+    try {
+      review.username = Meteor.user().username || Meteor.user().emails[0].address;
+      review.dateCreated = new Date;
+      Meteor.call("insert/review", review, function (error, result) {
+        console.log(error, result);
```
Comment (mradeybee) [inline imports/plugins/included/product-detail-simple/client/templates/review.js]: Why are you logging result here?

File: server/publications/collections/reviews.js (hunk 4 above)
Comment (louisdante9) [reply server/publications/collections/reviews.js]: addded

**Review by mradeybee — APPROVED** [review]
GTM

**Pull request merged**